### ISSUE TITLE
[1.16.x] Unable to connect to certain servers

### ISSUE THREAD

**Issue by llGaetanll** [sp614x/optifine#4667]
## Description of Issue
when using either `OptiFine 1.16.1 HD U G2` or  `OptiFine 1.16.2 HD U G3 pre2`, I find myself unable to join certain Minecraft servers. The error I get is the following
```
io.netty.channel.AbstractChannel$AnnotatedConnectException: syscall:getsockopt(..) failed: Connection refused: [ip omitted for obvious privacy reasons].apexmc.co/[ip here]:25565
    at io.netty.channel.unix.Socket.finishConnect(..)(Unknown Source) ~[Socket.class:4.1.25.Final]
Caused by: io.netty.channel.unix.Errors$NativeConnectException: syscall:getsockopt(..) failed: Connection refused
    ... 1 more
```

## Steps to Reproduce
1. Load Optifine 1.16.X
2. Connect to server

## OptiFine Version
`OptiFine 1.16.1 HD U G2` and `OptiFine 1.16.2 HD U G3 pre2` both have this issue. 
**Note that this issue does not occur on versions of Optifine below 1.16**

## OS/System Information
I added this section since I reckon it is relevant (My friends who use windows have no problem connecting to the server with Optifine)
```
OptiFine_1.16.2_HD_U_G3_pre2
Build: 20200820-215433
OS: Linux (amd64) version 5.8.1-arch1-1
Java: 14.0.2, N/A
VM: OpenJDK 64-Bit Server VM (mixed mode), Oracle Corporation
LWJGL: 3.3.0 X11 GLX EGL OSMesa clock_gettime evdev shared
OpenGL: GeForce GTX 1080/PCIe/SSE2, version 4.6.0 NVIDIA 450.57, NVIDIA Corporation
OpenGL Version: 4.6.0
```

## Installation Method
1. Downloaded Optifine from `https://optifine.net/downloads`
2. Ran `java -jar [optifine jarfile].jar` (so yea, basically I used the installer)

## Log Files/Crash Reports
As mentioned in the description above, relevant logs are
```
io.netty.channel.AbstractChannel$AnnotatedConnectException: syscall:getsockopt(..) failed: Connection refused: [ip omitted for obvious privacy reasons].apexmc.co/[ip here]:25565
    at io.netty.channel.unix.Socket.finishConnect(..)(Unknown Source) ~[Socket.class:4.1.25.Final]
Caused by: io.netty.channel.unix.Errors$NativeConnectException: syscall:getsockopt(..) failed: Connection refused
    ... 1 more
```

## F3 Debug Screenshot
N/A

## Prior Testing
If you've installed standalone:
- Does this happen in Vanilla without OptiFine?
Nope. I am able to join every server in vanilla Minecraft.

## Additional Information
I have opened an issue with the particular server host with which this issue occurs. We eventually came to a realization that this was most likely an Optifine issue, since I was able to connect to their servers without it just fine. Additionally, pinging their ip directly through my machine with `tracepath` resulted in a successful connection. Note that I am still unsure which servers have this problem, and the logs (at least to me) dont seem to clarify this at all. hopefully this issue can be resolved soon :)

**Comment by musabagriyanik** [comment]
> This particular issue is fixed by this PR.
> [sp614x/LegacyLauncher#1](https://github.com/sp614x/LegacyLauncher/pull/1)

is it added to the original optifine?

**Comment by SecretlyJealous** [comment]
It's because of: https://github.com/sp614x/optifine/issues/2917, sp614x/LegacyLauncher#1 may fix the issue but it has yet to be added to Optifine.

**Comment by sp614x** [comment]
Fixed in 1.16.2 pre4
